src/main/java/org/apache/sling/resourceresolver/impl/CommonResourceResolverFactoryImpl.java

```diff
@@ -472,6 +466,19 @@ public boolean hasVanityPathPrecedence() {
         return configs;
     }
 
+    @Override
+
+    public CopyOnWriteArrayList<String> getAllowedAliasPaths() {
```

Comment:
This should be: ```
public List<String> getAllowedAliasPaths() {
   return Collections.unmodifiableSet(this.aliasPathAllowList);
}
```

Reply:
@wimsymons 
the idea to return an immutable list is every time the unmodifiable list will be returned and not affected by configuration changes.  After getting all allowed list we need to sort the paths in (` public List<String> getAllowedAliasPaths() ` ), this way we are creating a modifiable list once again
`  List<String> sortedList = immutablelist.stream().sorted().collect(Collectors.toList());`

On other hand sorting paths in the activate method will not be advisable as the activate method should do the minimum required work.

Please suggest

Reply:
I suggest that sorting a list with 3 +/- 2 entries is acceptable in the activate method. This "performance hit" is fully compensated by not needing to sort the list repeatedly later on. Come on!